src/libraries/Microsoft.Extensions.Configuration.Binder/gen/ConfigurationBindingGenerator.Parser.cs

```diff
@@ -78,6 +80,27 @@ private static bool IsValidRootConfigType(ITypeSymbol? type)
                 }
 
                 return true;
+
+                static bool IsAccessibleFromGenBinders(ITypeSymbol type)
```

Comment:
Couldn't you use `IsSymbolAccessibleWithin` instead?

https://learn.microsoft.com/en-us/dotnet/api/microsoft.codeanalysis.compilation.issymbolaccessiblewithin?view=roslyn-dotnet-4.6.0

Reply:
I don't have a "within" symbol. The symbol would be the generated binder extension class which doesn't exist yet.

Reply:
Couldn't you use another type symbol in the same assembly as a proxy for the one that will be generated?

Reply:
It should be possible to use the current IAsse blySymbol as the 'within' parameter although I haven't tested that myself.

Reply:
Yes I considered the proxy approach. It didn't feel deterministic on first thought, but yes it would be a correct/better check. I'll also try using the assembly symbol.

Reply:
Using a proxy would work in the common case, but we'd need to account for the edge case of the assembly having just one (e.g. a singular, simple target-config POCO with primitive fields). Any handwritten fallback would have a vastly different impl.

Comparing with assembly doesn't work - nested privates aren't visible.

Reply:
> Comparing with assembly doesn't work - nested privates aren't visible.

Are any of the types being generated nested within user-defined types? If not, I would expect its visibility to be equivalent to that of the assembly?

Reply:
> Comparing with assembly doesn't work - nested privates aren't visible.

I tried it again; the assembly check works. Must have not used the `!` operator when I tried. Thanks.